application/shared-kernel/InfrastructureCore/InfrastructureCoreConfiguration.cs

```diff
@@ -11,75 +11,36 @@ namespace PlatformPlatform.SharedKernel.InfrastructureCore;
 
 public static class InfrastructureCoreConfiguration
 {
-    private static string? _cachedConnectionString;
-
     public static readonly bool SwaggerGenerator = Environment.GetEnvironmentVariable("SWAGGER_GENERATOR") == "true";
 
     [UsedImplicitly]
-    public static IServiceCollection ConfigureInfrastructureCoreServices<T>(
+    public static IServiceCollection ConfigureDatabaseContext<T>(
         this IServiceCollection services,
-        IConfiguration configuration,
-        Assembly assembly
+        IHostApplicationBuilder builder,
+        string connectionName
     ) where T : DbContext
     {
-        services.ConfigureDatabaseContext<T>(configuration);
-
-        services.RegisterRepositories(assembly);
+        builder.AddSqlServerDbContext<T>(connectionName,
+            static settings => settings.DbContextPooling = false);
 
         return services;
     }
 
     [UsedImplicitly]
-    private static IServiceCollection ConfigureDatabaseContext<T>(
+    public static IServiceCollection ConfigureInfrastructureCoreServices<T>(
         this IServiceCollection services,
-        IConfiguration configuration
+        Assembly assembly
```

Comment:
`assembly` should be called `infrastrctureAssembly` (to align with it being called `applicationAssembly` in `ApplicationCoreConfiguration`. I realise this was a problem before too. :)